### PR TITLE
Add mobile touch events

### PR DESCRIPTION
**Pull request by lee101**
Adds support for adding touch events "touchstart touchmove touchend".

Events can be added with `L.DomEvent.addListener(map, "touchstart", touchStartHandlerFunction);`

I found the need for these events in leaflet while developing a drag and drop "orange man" style leaflet control so people could select a point on the map.

Its interesting to note that that touchend isn't the same as mouseup, as i understand it mouseup fires on the element where the mouse was released and touchend fires on the element where touchstart occurred.

Something strange i noticed was that touchend isn't propagated if touchstart's on leaflet controls so i needed to attach a touchend event to the leaflet control and figure out the lat,lng using `map.mouseEventToLatLng(event)` (as oppose to attaching touchend to the map and doing `event.latlng`)

a shim to add them is described in #1542  
also related to #3075

### CODE REVIEW

**Comment by mourner** [comment]
Sorry for a late response! I've been meaning to add something like this a long time ago. Need some time to think about this though.

File: src/dom/DomEvent.js [hunk 1]
```diff
@@ -172,15 +172,19 @@ L.DomEvent = {
 	},
 
 	getMousePosition: function (e, container) {
+		var evt = e;
+		if (evt.touches) {
+			evt = evt.touches[0] || evt.changedTouches[0];
```
Comment (IvanSanchez) [inline src/dom/DomEvent.js]: I don't think this is a good idea. After all, the function is called get**Mouse**Position and not getEventPosition.

I presume there will be edge cases where the first touch originates outside the map container, then a second one inside, but `getMousePosition` uses only the first one.

I would argue that a new getTouchPosition**s** function is needed, to translate all the touch points into map coordinates.

Until then, it should be fine to expose just `ev.originalEvents` (as with `keypress` events, see `L.Map._fireDomEvent()`) and let the upper functionality decide which touches (if any) should get more attention and if whether to do some `containerPointToLatLng` calls or whatever.

Comment (IvanSanchez) [inline src/dom/DomEvent.js]: Or maybe make sub-cases of `_fireDOMEvent` named `_fireMouseEvent`, `_fireTouchEvent` and so on.

**Comment by alexkavon** [comment]
Perhaps utilize something like hammerjs?

**Comment by IvanSanchez** [comment]
> Perhaps utilize something like hammerjs?

Not sure that's a good idea either. HammerJS might interfere with `Map.Tap.js` and alternative libraries used by developers including Leaflet in their webpages.

---

I'm not convinced by this PR - I think managing coordinate transformations of multi-touch events has to be more complete, and AFAIK one-touch events are emulated properly as mouse drags.

**Comment by IvanSanchez** [comment]
After the `PointerEvent` updates in chrome 55, I don't think this is worth anymore. A better idea would be to switch everything to `PointerEvent` anyway.

**Comment by alexkavon** [comment]
That wouldn't really be friendly to iOS devices...though perhaps I'm misunderstanding?

http://caniuse.com/#feat=pointer

**Pull request closed without merging**